tests/attn_scores/test_attn_scores.py

```
@@ -0,0 +1,781 @@
+    context_lens = torch.full((batch_size,), avg_ctx, dtype=torch.int32, device=device)
+    block_table, num_total_blocks = _make_paged_kv(batch_size, phys_block_kv, context_lens, device)
+    q_f32 = torch.randn(batch_size, next_n, num_heads, head_dim, device=device, dtype=torch.bfloat16)
+    kv_cache = torch.randn(num_total_blocks, phys_block_kv, 1, head_dim, device=device, dtype=torch.bfloat16)
+    weights = torch.randn(batch_size * next_n, num_heads, device=device, dtype=torch.float32)
+
+    q_packed, sf_q_packed = _per_token_cast_to_fp4(q_f32.view(-1, head_dim), gran_k=32)
+    q_fp4 = q_packed.view(torch.uint8).view(batch_size, next_n, num_heads, head_dim // 2)
+    sf_q = sf_q_packed.view(torch.int32).view(batch_size, next_n, num_heads)
+
+    # Pre-transposed KV SF
+    kv_fused_transposed, kv_sim = _kv_cache_cast_to_fp4(
```

Ruff (0.16.1)

[warning] Unpacked variable `kv_sim` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)